app/app/upload-non-cr3/page.tsx

```diff
@@ -0,0 +1,257 @@
+
+  const onSelectFile = useCallback(
+    (file: File) => {
+      Papa.parse<NonCr3Upload>(file, {
+        delimiter: ",",
+        header: true,
+        skipEmptyLines: true,
+        complete: (results: ParseResult<NonCr3Upload>) => {
+          if (results.errors.length > 0) {
+            // handle CSV parsing errors
+            const formattedErrors: NonCr3ValidationError[] = results.errors.map(
+              (issue) => ({
+                rowNumber: (issue.row || 0) + 1,
+                fieldName: "",
+                message: issue.message,
+              })
+            );
+            setValidationErrors(formattedErrors);
+          } else {
+            // CSV has been parsed — run schema validations
+            console.log(results.data);
+            try {
+              const parsedData: NonCr3Upload[] =
+                NonCr3UploadDedupedSchema.parse(results.data);
+              setData(parsedData);
+            } catch (err) {
+              if (err instanceof ZodError) {
+                const formattedErrors = err.issues.map((issue) => {
+                  const [rowNumber, fieldName] = issue.path;
+                  const message = issue.message;
+                  return {
+                    rowNumber: Number(rowNumber) + 1,
+                    fieldName: String(fieldName),
+                    message,
+                  };
+                });
+                setValidationErrors(formattedErrors);
+              }
+            }
+          }
+          setParsing(false);
+        },
+      });
+    },
+    [setValidationErrors, setParsing, setData]
+  );
+
+  const onUpload = async () => {
+    try {
+      await mutate({ objects: data });
+      setSuccess(true);
+    } catch (err) {
+      console.error(err);
+      setUploadError(true);
+      setSuccess(false);
+    }
+  };
+
+  return (
+    <>
+      <AppBreadCrumb />
+      <Card>
+        <Card.Header className="fs-5 fw-bold">
+          Upload Non-CR3 records
+        </Card.Header>
+        <Card.Body>
+          {!data && (
+            <Row>
+              <Col xs={12} md={6} lg={3}>
+                <Form onSubmit={(e) => e.preventDefault()}>
+                  <Form.Control
+                    type="file"
+                    name="file"
+                    accept=".csv"
+                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                      if (validationErrors) {
+                        setValidationErrors(null);
+                      }
+                      if (e.target?.files) {
+                        setSuccess(false);
+                        setParsing(true);
+                        onSelectFile(e.target?.files[0]);
+                      }
+                    }}
+                  ></Form.Control>
+                </Form>
+              </Col>
+              <Col className="my-auto">
+                <Link
+                  href="/files/non_cr3_template.csv"
```

Comment:
@johnclary this file is missing from the public folder but I was able to test with the template from staging. 🥼

edit: I just realized that .csvs are gitignored at the root project level so that makes sense why it snuck out of this diff. I was so confused when I was seeing a change locally. 😵‍💫